### PR TITLE
init model

### PR DESCRIPTION
**Pull request by tuanhnt1712**

### CODE REVIEW

**Comment by tuanhnt1712** [comment]
Framgia CI Report Tool 0.2.1

------------------------------------------
Running Test
------------------------------------------

[+] Running:  rails_best_practices -e "db/schema.rb,db/migrate,vendor,app/models/application_record.rb" --format html --output-file .framgia-ci-reports/rails_best_practices.html app/
Source Code: |================================================================================================================================|
[+] Running:  bundle exec rubocop --require rubocop/formatter/checkstyle_formatter --format RuboCop::Formatter::CheckstyleFormatter --no-color --rails --out .framgia-ci-reports/rubocop.xml app/ lib/
[+] Running:  scss-lint --require=scss_lint_reporter_checkstyle --format=Checkstyle --out=.framgia-ci-reports/scss-lint.xml app/assets/stylesheets/
[+] Running:  eslint --format=checkstyle --output-file=.framgia-ci-reports/eslint.xml app/
[+] Running:  reek --format html > .framgia-ci-reports/reek.html
[+] Running:  brakeman -o .framgia-ci-reports/brakeman.html -o .framgia-ci-reports/brakeman.json
Loading scanner...
Processing application in /home/tuanh/workspace/framgia-project2
Processing gems...
[Notice] Detected Rails 5 application
Processing configuration...
[Notice] Escaping HTML by default
Parsing files...
Processing initializers...
Processing libs...ed
Processing routes...          
Processing templates...       
Processing data flow in templates...
Processing models...          
Processing controllers...     
Processing data flow in controllers...
Indexing call sites...        
Running checks in parallel...
 - CheckBasicAuth
 - CheckBasicAuthTimingAttack
 - CheckCrossSiteScripting
 - CheckContentTag
 - CheckCreateWith
 - CheckDefaultRoutes
 - CheckDeserialize
 - CheckDetailedExceptions
 - CheckDigestDoS
 - CheckDynamicFinders
 - CheckEscapeFunction
 - CheckEvaluation
 - CheckExecute
 - CheckFileAccess
 - CheckFileDisclosure
 - CheckFilterSkipping
 - CheckForgerySetting
 - CheckHeaderDoS
 - CheckI18nXSS
 - CheckJRubyXML
 - CheckJSONEncoding
 - CheckJSONParsing
 - CheckLinkTo
 - CheckLinkToHref
 - CheckMailTo
 - CheckMassAssignment
 - CheckMimeTypeDoS
 - CheckModelAttrAccessible
 - CheckModelAttributes
 - CheckModelSerialize
 - CheckNestedAttributes
 - CheckNestedAttributesBypass
 - CheckNumberToCurrency
 - CheckQuoteTableName
 - CheckRedirect
 - CheckRegexDoS
 - CheckRender
 - CheckRenderDoS
 - CheckRenderInline
 - CheckResponseSplitting
 - CheckRouteDoS
 - CheckSafeBufferManipulation
 - CheckSanitizeMethods
 - CheckSelectTag
 - CheckSelectVulnerability
 - CheckSend
 - CheckSendFile
 - CheckSessionManipulation
 - CheckSessionSettings
 - CheckSimpleFormat
 - CheckSingleQuotes
 - CheckSkipBeforeFilter
 - CheckSQL
 - CheckSQLCVEs
 - CheckSSLVerify
 - CheckStripTags
 - CheckSymbolDoSCVE
 - CheckTranslateBug
 - CheckUnsafeReflection
 - CheckValidationRegex
 - CheckWithoutProtection
 - CheckXMLDoS
 - CheckYAMLParsing
Checks finished, collecting results...
Generating report...
Report saved in '.framgia-ci-reports/brakeman.html'
Report saved in '.framgia-ci-reports/brakeman.json'
[+] Running:  rspec --format html --out .framgia-ci-reports/rspec.html spec/

An error occurred while loading ./spec/models/comment_spec.rb.
Failure/Error: require File.expand_path("../../config/environment", __FILE__)

LoadError:
  /tmp/_MEIfuPTDi/libssl.so.1.0.0: version `OPENSSL_1.0.2' not found (required by /home/tuanh/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/x86_64-linux/openssl.so) - /home/tuanh/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/x86_64-linux/openssl.so
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.4/lib/active_support/key_generator.rb:2:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.4/lib/active_support/key_generator.rb:2:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/application.rb:4:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/application.rb:4:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails.rb:12:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails.rb:12:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/all.rb:1:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/all.rb:1:in `<top (required)>'
# ./config/application.rb:3:in `require'
# ./config/application.rb:3:in `<top (required)>'
# ./config/environment.rb:2:in `require_relative'
# ./config/environment.rb:2:in `<top (required)>'
# ./spec/rails_helper.rb:3:in `require'
# ./spec/rails_helper.rb:3:in `<top (required)>'
# ./spec/models/comment_spec.rb:1:in `require'
# ./spec/models/comment_spec.rb:1:in `<top (required)>'

An error occurred while loading ./spec/models/post_spec.rb.
Failure/Error: require File.expand_path("../../config/environment", __FILE__)

LoadError:
  /tmp/_MEIfuPTDi/libssl.so.1.0.0: version `OPENSSL_1.0.2' not found (required by /home/tuanh/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/x86_64-linux/openssl.so) - /home/tuanh/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/x86_64-linux/openssl.so
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.4/lib/active_support/key_generator.rb:2:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.4/lib/active_support/key_generator.rb:2:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/application.rb:4:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/application.rb:4:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails.rb:12:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails.rb:12:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/all.rb:1:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/all.rb:1:in `<top (required)>'
# ./config/application.rb:3:in `require'
# ./config/application.rb:3:in `<top (required)>'
# ./config/environment.rb:2:in `require_relative'
# ./config/environment.rb:2:in `<top (required)>'
# ./spec/rails_helper.rb:3:in `require'
# ./spec/rails_helper.rb:3:in `<top (required)>'
# ./spec/models/post_spec.rb:1:in `require'
# ./spec/models/post_spec.rb:1:in `<top (required)>'

An error occurred while loading ./spec/models/post_tag_spec.rb.
Failure/Error: require File.expand_path("../../config/environment", __FILE__)

LoadError:
  /tmp/_MEIfuPTDi/libssl.so.1.0.0: version `OPENSSL_1.0.2' not found (required by /home/tuanh/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/x86_64-linux/openssl.so) - /home/tuanh/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/x86_64-linux/openssl.so
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.4/lib/active_support/key_generator.rb:2:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.4/lib/active_support/key_generator.rb:2:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/application.rb:4:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/application.rb:4:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails.rb:12:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails.rb:12:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/all.rb:1:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/all.rb:1:in `<top (required)>'
# ./config/application.rb:3:in `require'
# ./config/application.rb:3:in `<top (required)>'
# ./config/environment.rb:2:in `require_relative'
# ./config/environment.rb:2:in `<top (required)>'
# ./spec/rails_helper.rb:3:in `require'
# ./spec/rails_helper.rb:3:in `<top (required)>'
# ./spec/models/post_tag_spec.rb:1:in `require'
# ./spec/models/post_tag_spec.rb:1:in `<top (required)>'

An error occurred while loading ./spec/models/relationship_spec.rb.
Failure/Error: require File.expand_path("../../config/environment", __FILE__)

LoadError:
  /tmp/_MEIfuPTDi/libssl.so.1.0.0: version `OPENSSL_1.0.2' not found (required by /home/tuanh/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/x86_64-linux/openssl.so) - /home/tuanh/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/x86_64-linux/openssl.so
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.4/lib/active_support/key_generator.rb:2:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.4/lib/active_support/key_generator.rb:2:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/application.rb:4:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/application.rb:4:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails.rb:12:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails.rb:12:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/all.rb:1:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/all.rb:1:in `<top (required)>'
# ./config/application.rb:3:in `require'
# ./config/application.rb:3:in `<top (required)>'
# ./config/environment.rb:2:in `require_relative'
# ./config/environment.rb:2:in `<top (required)>'
# ./spec/rails_helper.rb:3:in `require'
# ./spec/rails_helper.rb:3:in `<top (required)>'
# ./spec/models/relationship_spec.rb:1:in `require'
# ./spec/models/relationship_spec.rb:1:in `<top (required)>'

An error occurred while loading ./spec/models/tag_spec.rb.
Failure/Error: require File.expand_path("../../config/environment", __FILE__)

LoadError:
  /tmp/_MEIfuPTDi/libssl.so.1.0.0: version `OPENSSL_1.0.2' not found (required by /home/tuanh/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/x86_64-linux/openssl.so) - /home/tuanh/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/x86_64-linux/openssl.so
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.4/lib/active_support/key_generator.rb:2:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.4/lib/active_support/key_generator.rb:2:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/application.rb:4:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/application.rb:4:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails.rb:12:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails.rb:12:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/all.rb:1:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/all.rb:1:in `<top (required)>'
# ./config/application.rb:3:in `require'
# ./config/application.rb:3:in `<top (required)>'
# ./config/environment.rb:2:in `require_relative'
# ./config/environment.rb:2:in `<top (required)>'
# ./spec/rails_helper.rb:3:in `require'
# ./spec/rails_helper.rb:3:in `<top (required)>'
# ./spec/models/tag_spec.rb:1:in `require'
# ./spec/models/tag_spec.rb:1:in `<top (required)>'

An error occurred while loading ./spec/models/user_spec.rb.
Failure/Error: require File.expand_path("../../config/environment", __FILE__)

LoadError:
  /tmp/_MEIfuPTDi/libssl.so.1.0.0: version `OPENSSL_1.0.2' not found (required by /home/tuanh/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/x86_64-linux/openssl.so) - /home/tuanh/.rvm/rubies/ruby-2.4.0/lib/ruby/2.4.0/x86_64-linux/openssl.so
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.4/lib/active_support/key_generator.rb:2:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/activesupport-5.0.4/lib/active_support/key_generator.rb:2:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/application.rb:4:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/application.rb:4:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails.rb:12:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails.rb:12:in `<top (required)>'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/all.rb:1:in `require'
# /home/tuanh/.rvm/gems/ruby-2.4.0/gems/railties-5.0.4/lib/rails/all.rb:1:in `<top (required)>'
# ./config/application.rb:3:in `require'
# ./config/application.rb:3:in `<top (required)>'
# ./config/environment.rb:2:in `require_relative'
# ./config/environment.rb:2:in `<top (required)>'
# ./spec/rails_helper.rb:3:in `require'
# ./spec/rails_helper.rb:3:in `<top (required)>'
# ./spec/models/user_spec.rb:1:in `require'
# ./spec/models/user_spec.rb:1:in `<top (required)>'
No examples found.
[+] Running:  bundle-audit check --update | tee .framgia-ci-reports/bundle-audit.txt
Updating ruby-advisory-db ...
git-remote-https: relocation error: /usr/lib/x86_64-linux-gnu/libgnutls.so.30: symbol asn1_der_decoding2, version LIBTASN1_0_3 not defined in file libtasn1.so.6 with link time reference
Failed updating ruby-advisory-db!

------------------------------------------
[x] Build Fail!
------------------------------------------

[o] rails_best_practices: success
[o] rubocop: success
[o] scss-lint: success
[o] eslint: success
[o] reek: success
[o] brakeman: success
[x] rspec: failed
[o] bundle-audit: success

**Comment by tuanhnt1712** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+    :recoverable, :rememberable, :trackable, :validatable
+
+  enum role: [:guest, :admin]
```
Comment (dieunb) [inline app/models/user.rb:5]: thay vì `guest` thì để là `normal` sẽ hợp lý hơn

**Comment by dieunb** [comment]
merged-có 1 comment nhỏ, em xem rồi xử lý trong pull tới

**Pull request merged**